library/src/main/cpp/decoder_heif.cpp

```diff
@@ -51,35 +50,108 @@ ImageInfo HeifDecoder::parseInfo() {
   };
 }
 
+cmsHPROFILE HeifDecoder::getColorProfile(heif::ImageHandle handle) {
+  auto im_handle = handle.get_raw_image_handle();
+  size_t icc_size = heif_image_handle_get_raw_color_profile_size(im_handle);
+  if (icc_size > 0) {
+    std::vector<uint8_t> icc_profile(icc_size);
+    heif_image_handle_get_raw_color_profile(im_handle, icc_profile.data());
+
+    cmsHPROFILE src_profile =
+        cmsOpenProfileFromMem(icc_profile.data(), icc_size);
+    cmsColorSpaceSignature profileSpace = cmsGetColorSpace(src_profile);
+    int chromabits = handle.get_chroma_bits_per_pixel();
+
+    if (profileSpace != cmsSigRgbData &&
+        (chromabits > 0 || profileSpace != cmsSigGrayData)) {
+      cmsCloseProfile(src_profile);
+      return nullptr;
+    }
+
+    return src_profile;
+  } else {
+    return cmsCreate_sRGBProfile();
+  }
+}
+
 void HeifDecoder::decode(uint8_t* outPixels, Rect outRect, Rect inRect,
-                         bool rgb565, uint32_t sampleSize) {
+                         bool rgb565, uint32_t sampleSize,
+                         cmsHPROFILE target_profile) {
```

Comment:
Nitpick: inconsistent casing of `target_profile` compared to other arguments.

Reply:
Done.